shark_turbine/kernel/wave/codegen.py

```diff
@@ -368,6 +368,9 @@ def _construct_gather_scatter_indices(
         ), "non-identity input mapping is not supported yet"
         index_mapping = mapping.map_output_indices(symbolc_shape)
 
+    idxc = IndexingContext.current()
+    index_mapping = tuple(i.subs(idxc.subs) for i in index_mapping)
```

Comment:
For my own education, do you have example on how index mapping used to be and how it is with this change?

Reply:
as in what gets subbed during this index mapping, the tile sizes? haha

Reply:
```
Before: (floor($index0/((H - HF + 1)*(W - WF + 1))), floor($index1/(HF*WF)), floor((Mod($index1, HF*WF))/WF) + floor((Mod($index0, (H - HF + 1)*(W - WF + 1)))/(W - WF + 1)), Mod($index0, W - WF + 1) + Mod($index1, WF))
After:  (floor($index0/64), floor($index1/4), floor((Mod($index0, 64))/8) + floor((Mod($index1, 4))/2), Mod($index0, 8) + Mod($index1, 2))
```
Mapping can contain arbitrary symbols and we need to substitute them. Previous transpose example worked without it, because it was only using iterators (`$indexN`) and not any other symbols.

Reply:
Makes sense yeah! :)